local-storage-service/server/src/main/java/org/verapdf/webapp/localstorageservice/server/service/LocalFileService.java

```diff
@@ -54,13 +55,13 @@ private File getFileOnDiskByPath(String localPath) throws FileNotFoundException
 		return res;
 	}
 
-	private String saveFile(InputStream inputStream, String fileName,
-	                        String expectedContentMD5) throws VeraPDFBackendException, IOException {
+	private Pair<String, String> saveFile(InputStream inputStream, String fileName,
+	                                      String expectedContentMD5) throws VeraPDFBackendException, IOException {
 		File dirToSaveFile = getDirToSaveFile();
 		File toSave = new File(dirToSaveFile, fileName);
+		String actualContentMD5 = FilesTool.saveFileOnDiskAndCheck(inputStream, toSave, expectedContentMD5);
 
-		FilesTool.saveFileOnDiskAndCheck(inputStream, toSave, expectedContentMD5);
-		return this.fileBaseDir.toPath().relativize(toSave.toPath()).toString();
+		return Pair.of(actualContentMD5, this.fileBaseDir.toPath().relativize(toSave.toPath()).toString());
```

Comment:
it would be better to use file path as a first element of the pair and checksum as the 2nd